### ISSUE TITLE
tikv panic with "no such file or directory" when disk space is not enough

### ISSUE THREAD

**Issue by cosven** [tikv/tikv#10688]
### What version of TiKV are you using?
```
[root@centos76_vm tikv-20161]# bin/tikv-server -V
TiKV
Release Version:   5.1.0-alpha
Edition:           Community
Git Commit Hash:   353897753d27fedc6343a0b84bcab454cdec10d3
Git Commit Branch: master
UTC Build Time:    2021-07-27 09:39:00
Rust Version:      rustc 1.53.0-nightly (16bf626a3 2021-04-14)
Enable Features:   jemalloc mem-profiling portable sse protobuf-codec test-engines-rocksdb cloud-aws cloud-gcp
Profile:           dist_release
```

### Steps to reproduce

1. disk full and tikv restart repeatly with error log
```
[2021/08/09 12:05:06.956 +08:00] [FATAL] [server.rs:1177] ["failed to create raft engine: Storage Engine IO error: No space left on deviceWhile appending to file: /tidb/eindex/data/tikv-20161/       raft/226776.sst: No space left on device"]
```
2. remove the space_hodler_file manually

### What did you expect?

Don't know. Maybe restart with some error message, instead of panic with "No such file or directory".

### What did happened?
1. tikv panic with following log. 
```
called `Result::unwrap()` on an `Err` value: Os { code: 2, kind: NotFound, message: "No such file or directory" }"] [backtrace="stack backtrace:
   0: tikv_util::set_panic_hook::{{closure}}
             at components/tikv_util/src/lib.rs:462
   1: std::panicking::rust_panic_with_hook
             at library/std/src/panicking.rs:595
   2: std::panicking::begin_panic_handler::{{closure}}
             at library/std/src/panicking.rs:497
   3: std::sys_common::backtrace::__rust_end_short_backtrace
             at library/std/src/sys_common/backtrace.rs:141
   4: rust_begin_unwind
             at library/std/src/panicking.rs:493
   5: core::panicking::panic_fmt
             at library/core/src/panicking.rs:92
   6: core::result::unwrap_failed
             at library/core/src/result.rs:1355
   7: core::result::Result<T,E>::unwrap
             at /rustc/16bf626a31cb5b121d0bca2baa969b4f67eb0dab/library/core/src/result.rs:1037
      server::server::TiKVServer<ER>::init_fs
             at components/server/src/server.rs:416
      server::server::run_tikv
             at components/server/src/server.rs:151
   8: tikv_server::main
             at cmd/tikv-server/src/main.rs:181
   9: core::ops::function::FnOnce::call_once
             at /rustc/16bf626a31cb5b121d0bca2baa969b4f67eb0dab/library/core/src/ops/function.rs:227
      std::sys_common::backtrace::__rust_begin_short_backtrace
             at /rustc/16bf626a31cb5b121d0bca2baa969b4f67eb0dab/library/std/src/sys_common/backtrace.rs:125
  10: main
  11: __libc_start_main
  12: <unknown>
```
2. I checked the `space_holder_file` file size, and it is 0, which seems also strange.
```
[root@centos76_vm tikv-20161]# du -hs /tidb/eindex/data/tikv-20161/space_placeholder_file
0	/tidb/eindex/data/tikv-20161/space_placeholder_file
```

**Comment by 7yyo** [comment]
I have also encountered this problem, I hope this error message can be clearer

**Comment by zhangjinpeng1987** [comment]
@hicqu Does this issue fixed in master and 5.3?

**Comment by hicqu** [comment]
It should be fixed in master and 5.3.0. @7yyo would you mind to send another issue to describe it detailly? Version and topology will be helpful for us. Thank you!

**Comment by cosven** [comment]
> send another issue to describe it deta

@hicqu  Which PR fixes this issue? I want to know some details about this issue.

**Comment by hicqu** [comment]
The feature "disk full resolver" introduced in 5.3.0 should fix the problem. The feature is implemented in several PRs.

And, the feature doesn't handle panicks exactly because it can panic anywhere if disk usage reaches full. The feature just rejects write requests when disk usage reaches a threshold.

**Comment by hicqu** [comment]
@tier-cap can you give more information about this feature? I guess we need a PR or commit to close the issue.

**Comment by 7yyo** [comment]
> It should be fixed in master and 5.3.0. @7yyo would you mind to send another issue to describe it detailly? Or share more infomation about version and topology here. They will be helpful for us. Thank you!

Deploy the TiDB cluster, the disk size > `reserve-space`

**Comment by tier-cap** [comment]
Cannot re-produce it, and maybe we can close it by https://github.com/tikv/tikv/pull/10833.
